### ISSUE TITLE
[Xcode 10] Converting non-escaping value to 'Any' may allow it to escape

### ISSUE THREAD

**Issue by kdawgwilk** [MakeAWishFoundation/SwiftyMocky#124]
All of the Perform static methods generated are failing to compile in Xcode 10 beta 4 with the following error:
```
Converting non-escaping value to 'Any' may allow it to escape
```

I realize that Xcode 10 is a beta and so I wouldn't expect things to "just work" but wanted to bring it up here so that we are aware of the issue.  This will allow us to come up with some solutions throughout the beta period that can then be implemented when the final release comes.  Not sure if backward compatibility is going to be an issue we need to keep in mind as well.

**Comment by amichnia** [comment]
Hi @kdawgwilk, thanks for raising this issue. I'm back from holiday, so I'll start looking into it. Still - until Xcode 10 GM is released, I will expect to have additional beta branch.

**Comment by darecki** [comment]
The same happens on Xcode 10 beta 5.
@amichnia any progress on that?

**Comment by amichnia** [comment]
Hi, work has started, we already have pr and we opened version 3.x project. Sorry for so much delay :)

**Comment by draperj** [comment]
Is there a current beta branch with this fix in place that can be used? I've had to avoid using Xcode 10 beta and GM due to a number of our projects using SwiftyMocky.

**Comment by amichnia** [comment]
@draperj  If you use this branch, you should be able to use get_sourcery script. We already uploaded prebuilt binary for upcoming XCode version.

**Comment by draperj** [comment]
I can confirm this does work with the Xcode 10 GM! Steps I followed:
1. Adjusted to "pod 'SwiftyMocky', :git => 'https://github.com/MakeAWishFoundation/SwiftyMocky.git', :branch => 'release/3.0" in PodSpec
2. Run Pod install
3. CD to SwiftyMocky pod folder
4. run the "get_sourcery.sh" command
5. Rerun your swiftymocky watch command
6. Things should run as expected now!

**Comment by Przemyslaw-Wosko** [comment]
@kdawgwilk @draperj hmm, i fixed those issues a while ago, but seems that i forgot to inform you how to get new sourcery and branch with beta changes. 

Where should i inform community about fixes and changes? should i edit Readme from master branch? 
if you have any ideas, let me know :)

**Comment by draperj** [comment]
@Przemyslaw-Wosko Readme on master makes sense to me.

**Comment by draperj** [comment]
This seems to no longer be working for me, but I think it is related to the Sourcery dependency. When running the generate command I see the following:

> Pods/Sourcery/bin/sourcery --config mocky.yml
Using configuration file at 'mocky.yml'
Scanning sources...
Found 9 types.
Loading templates...
Loaded 1 templates.
Generating code...
error: Error Domain=NSOSStatusErrorDomain Code=-1 "kCFStreamErrorHTTPParseFailure / kCFSocketError / kCFStreamErrorDomainCustom / kCSIdentityUnknownAuthorityErr / qErr / telGenericError / dsNoExtsMacsBug / kMovieLoadStateError / cdevGenErr: Could not parse the request/response. /  / custom to the kind of stream in question  /  / queue element not found during deletion /  / not a SysErr, just a placeholder  /  / General error; gray cdev w/o alert" UserInfo={terminationReason=__ObjC.Process.TerminationReason, error=dyld: Symbol not found: __T0BOWV
  Referenced from: /Users/draperj/Library/Caches/Sourcery/Mock.swifttemplate/../SourceryRuntime.framework/Versions/A/SourceryRuntime
  Expected in: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/libswiftCore.dylib
 in /Users/draperj/Library/Caches/Sourcery/Mock.swifttemplate/../SourceryRuntime.framework/Versions/A/SourceryRuntime
}

**Comment by amichnia** [comment]
I will look into it, thanks for pointing this out

**Comment by amichnia** [comment]
@draperj - we are still dependent on new Sourcery version, I might take some time before things get ready I'm afraid. We will keep you guys updated

**Comment by amichnia** [comment]
@draperj  One more note. When using "get_sourcery.sh" script, you have to use a different path to the binary - `./Pods/Sourcery/bin/Sourcery.app/Contents/MacOS/Sourcery`

**Comment by amichnia** [comment]
Considering it done for now. As a temporary solution until 3.0 is released, please use `release/3.0` branch, and `get_sourcery.sh swift/4.2-xcode10`
